tests/ctest/test_keyset_cow.c

```diff
@@ -0,0 +1,349 @@
+/**
+* @file
+*
+* @brief
+*
+* @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+*/
+
+#include "kdbprivate.h"
+#include <tests_internal.h>
+
+static void ksBelow_root_in_keyset_should_work (void)
+{
+	printf("Test %s\n", __func__);
+
+	// Arrange
+	Key * root = keyNew ("system:/root", KS_END);
+
+	KeySet * ks = ksNew (KEYSET_SIZE, root, keyNew ("system:/root/abc", KEY_END), keyNew ("system:/other", KEY_END), KS_END);
+
+	// Act
+	KeySet * below = ksBelow (ks, root);
+
+	// Assert
+	succeed_if_fmt(ksGetSize (below) == 2, "should contain 2 keys, was %zd", ksGetSize (below));
+	succeed_if (ksLookupByName (below, "system:/root", 0) != NULL, "should contain root");
+	succeed_if (ksLookupByName (below, "system:/root/abc", 0) != NULL, "should contain key below root");
+	succeed_if (ksGetSize (ks) == 3, "all keys should remain in keyset");
+
+	keyDel (root);
+	ksDel (ks);
+	ksDel (below);
+}
+
+static void ksBelow_key_copied_from_root_in_keyset_should_work (void)
+{
+	printf("Test %s\n", __func__);
+
+	// Arrange
+	Key * root = keyNew ("system:/root", KS_END);
+	Key * copy = keyCopy (keyNew ("/", KEY_END), root, KEY_CP_ALL);
+
+	KeySet * ks = ksNew (KEYSET_SIZE, copy, keyNew ("system:/root/abc", KEY_END), keyNew ("system:/other", KEY_END), KS_END);
+
+	// Act
+	KeySet * below = ksBelow (ks, root);
+
+	// Assert
+	succeed_if_fmt(ksGetSize (below) == 2, "should contain 2 keys, was %zd", ksGetSize (below));
+	succeed_if (ksLookupByName (below, "system:/root", 0) != NULL, "should contain root");
+	succeed_if (ksLookupByName (below, "system:/root/abc", 0) != NULL, "should contain key below root");
+	succeed_if (ksGetSize (ks) == 3, "all keys should remain in keyset");
+
+	keyDel (root);
+	keyDel (copy);
+	ksDel (ks);
+	ksDel (below);
+}
+
+static void ksBelow_cascading_root_in_keyset_should_work (void)
+{
+	printf("Test %s\n", __func__);
+
+	// Arrange
+	Key * root = keyNew ("/root", KS_END);
+
+	KeySet * ks = ksNew (KEYSET_SIZE, root, keyNew ("system:/root/abc", KEY_END), keyNew ("system:/other", KEY_END), KS_END);
+
+	// Act
+	KeySet * below = ksBelow (ks, root);
+
+	// Assert
+	succeed_if_fmt(ksGetSize (below) == 2, "should contain 2 keys, was %zd", ksGetSize (below));
+	succeed_if (ksLookupByName (below, "/root", 0) != NULL, "should contain root");
```

Comment:
As stated in https://github.com/ElektraInitiative/libelektra/pull/4691#discussion_r1027871203, either this key shouldn't be there, or `/root/abc` should also be there, if you add it to `ks`. Please adapt this one and the other tests.

Reply:
All tests or only the tests with cascading keys?

Reply:
At least to the ones with a cascading `root`. For the other's it doesn't really matter, but it also wouldn't hurt.

Reply:
Done, added it to all testcases